tests/daemon/test-persistent-cache.c

```diff
@@ -1041,33 +1018,39 @@ test_persistent_cache_reads_cached_boot_offset (gboolean     *unused,
 }
 
 /*
- * Ensures that a request for the boot time offset doesn't cause a write to the
- * metadata file if the 'always_update_timestamps' parameter is set to FALSE.
+ * Tests that the in-memory cache is used when present by removing the boot
+ * offset from disk between calls to the same instance of a persistent cache.
+ * This test exercises a code path similar to
+ * test_persistent_cache_reads_cached_boot_offset_when_file_changed().
  */
 static void
-test_persistent_cache_get_offset_wont_update_timestamps_if_it_isnt_supposed_to (gboolean     *unused,
-                                                                                gconstpointer dontuseme)
+test_persistent_cache_reads_cached_boot_offset_when_file_missing (gboolean *unused,
+                                                                  gconstpointer dontuseme)
 {
   EmerPersistentCache *cache = make_testing_cache ();
 
-  gint64 relative_time = read_relative_time ();
-  gint64 absolute_time = read_absolute_time ();
-
-  // Make a little time pass.
-  g_usleep (75000); // 0.075 seconds
-
+  gint64 first_offset;
   GError *error = NULL;
-
-  // This call shouldn't update the metadata file.
   gboolean get_succeeded =
-    emer_persistent_cache_get_boot_time_offset (cache, NULL, &error);
+    emer_persistent_cache_get_boot_time_offset (cache, &first_offset, &error);
+  g_assert_no_error (error);
+  g_assert_true (get_succeeded);
+
+  // Remove the boot offset file.
+  g_assert_cmpint (g_unlink (TEST_DIRECTORY BOOT_OFFSET_METADATA_FILE), ==, 0);
 
+  gint64 second_offset;
+
+  /*
+   * This call should read the offset from its in-memory cache, not the new one
```

Comment:
Maybe just remove ", not the new one on disk" since that only applies in the case of the other test.